packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -171,15 +203,35 @@ export const GridRootStyles = styled('div', {
         t.palette.action.selectedOpacity + t.palette.action.hoverOpacity,
       );
 
-  const pinnedHoverBackground = t.vars
+  const pinnedBackgroundColor = t.vars
     ? hoverColor
     : blend(pinnedBackground, hoverColor, hoverOpacity);
-  const pinnedSelectedBackground = t.vars
+  const pinnedHoverStyles = getPinnedBackgroundStyles({
+    hoverOpacity,
+    selectedOpacity,
+    selectedBackground,
+    backgroundColor: pinnedBackgroundColor,
+  });
+
+  const pinnedSelectedBackgroundColor = t.vars
     ? selectedBackground
     : blend(pinnedBackground, selectedBackground, selectedOpacity);
-  const pinnedSelectedHoverBackground = t.vars
-    ? hoverColor
-    : blend(pinnedSelectedBackground, hoverColor, hoverOpacity);
+  const pinnedSelectedStyles = getPinnedBackgroundStyles({
+    hoverOpacity,
+    selectedOpacity,
+    selectedBackground,
+    backgroundColor: pinnedSelectedBackgroundColor,
```

Comment:
`pinnedSelectedBackgroundColor` here can be either a CSS color value or a CSS variable.
`getPinnedBackgroundStyles` assumes it's a CSS color value and uses the `blend` function, which won't work with CSS variables.
This made me realize that we missed a test for the DataGrid using inside of the `CssVarsProvider`, so I opened https://github.com/mui/mui-x/pull/14662 to add one.

We have to make sure we don't try to `blend` CSS variables.
This also reminds me about this PR: https://github.com/mui/mui-x/pull/12901.

Reply:
@cherniavskii I am not sure of the best way to handle this. 

`color-mix` would work well besides the versions of Safari that do not support it. It doesn't look like there is another way to blend colors with CSS variables otherwise.

I don't know if we can add an exception to our browser support specifically for cases where `CssVarsProvider` is used? 😅 

The only other way I can think to achieve this effect with CSS variables is by having an inner element in pinned cells to apply the transparent hover/selected background-color to. Not sure this is viable though, considering the performance implications of having an extra element for every pinned cell. https://codepen.io/KenanYusuf/pen/WNqVXqM

Reply:
@KenanYusuf, not sure it helps, but we ran into similar problems when migrating to DataGrid v7 due to our custom styles. Some of our pinned cells had backgrounds with alpha values. Therefore, the non-pinned cells were visible behind them.

We attempted to solve this using `color-mix`. The result worked but was very hard to read and extend in our case.

We opted for the two-element approach you described, but instead of adding one additional element per cell, we added one sticky background element with the width of the pinned cells.

<details><summary>Details</summary>
<p>

![grafik](https://github.com/user-attachments/assets/f5fbeaf2-22c9-484f-95e2-568bdc82c16b)

We added the sticky background to the row using DataGrid `slots`. We had to place it between row and pinned cells using `z-index`, but you could probably just make it a container element of the pinned cells.

The additional sticky background always has the same background color as the row. If the row is selected using row selection, the color of the additional sticky background changes too. If only a pinned cell is selected using cell selection, only the background color of the pinned cell changes.

</p>
</details>

We didn't run into noticeable performance problems. If you consider pinned left and pinned right, it's only two more divs in each row.